dataline-scheduler/src/main/java/io/dataline/scheduler/WorkerRunFactory.java

```diff
@@ -71,9 +72,11 @@ public WorkerRunFactory(final Path workspaceRoot,
   }
 
   public WorkerRun create(final Job job) {
-    LOGGER.info("job: {} {} {}", job.getId(), job.getScope(), job.getConfig().getConfigType());
+    LOGGER.info("job id: {} attempt: {} scope: {} type: {}", job.getId(), job.getAttempts(), job.getScope(), job.getConfig().getConfigType());
 
-    final Path jobRoot = workspaceRoot.resolve(String.valueOf(job.getId()));
+    final Path jobRoot = workspaceRoot.resolve(String.valueOf(job.getId())).resolve(String.valueOf(job.getAttempts()));
+    // make sure the workspace is empty.
+    FileUtils.deleteQuietly(jobRoot.toFile());
```

Comment:
why do you need that?

Reply:
This might hide a bug, we shouldn't protect ourselves against that.

Reply:
ok. if we reset a job that has failed too many times manually, then we need to remember to delete a bunch of stuff is all.

Reply:
Shouldn't that create a new job instead?